### ISSUE TITLE
NullPointerException in FileChangesProcessor

### ISSUE THREAD

**Issue by tobiasgeissler** [jenkinsci/code-coverage-api-plugin#487]
### Jenkins and plugins versions report
Jenkins Version 2.346.2
Code Coverage API Plugin Version 3.0.5

### Reproduction steps
1. Have a file within the repository in a folder which is not part of the coverage report. In our case it was a java class inside a testsrc folder.
2. Create a jenkins build based on this repository state.
3. Move the file to a folder which is part of the coverage report. In our case we moved the file to the src folder and also changed its package but kept the name of the class.
4. Create another jenkins build based on the new state where the first build is taken as a reference build for the publishCoverage action.

Note: We're not 100% sure if these are the only prerequisites to reproduce the exception. However, when we added the original file to the project again, the exception was not thrown anymore.

### Expected Results
No NullPointerException is thrown and the build is successful.

### Actual Results
The following NullPointerException is thrown and the build fails:
```
java.lang.NullPointerException
	at io.jenkins.plugins.coverage.model.FileChangesProcessor.attachFileCoverageDelta(FileChangesProcessor.java:98)
	at io.jenkins.plugins.coverage.model.FileChangesProcessor.lambda$attachFileCoverageDeltas$1(FileChangesProcessor.java:86)
	at java.base/java.util.HashMap.forEach(HashMap.java:1337)
	at io.jenkins.plugins.coverage.model.FileChangesProcessor.attachFileCoverageDeltas(FileChangesProcessor.java:85)
	at io.jenkins.plugins.coverage.model.CoverageReporter.run(CoverageReporter.java:110)
	at io.jenkins.plugins.coverage.CoverageProcessor.performCoverageReport(CoverageProcessor.java:133)
	at io.jenkins.plugins.coverage.CoveragePublisher.perform(CoveragePublisher.java:114)
	at jenkins.tasks.SimpleBuildStep.perform(SimpleBuildStep.java:123)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:101)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:71)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by uhafner** [comment]
Thanks for the detailed bug report! 

@fo-code: When looking at https://github.com/jenkinsci/code-coverage-api-plugin/blob/master/plugin/src/main/java/io/jenkins/plugins/coverage/model/FileChangesProcessor.java#L85 it seems that we  simply need to skip the computation of a delta for files that have no counter part in a previous report? Is this assumption correct or will it create problems in the post processing steps?

**Comment by fo-code** [comment]
@uhafner that is correct. Nonetheless, these files should not even be processed here. There is a bug when creating the mapping between former and current report paths - empty reference paths have been stored there.

I already added a fix for that - see #489 .
